lib/cocina/models/builders/title_builder.rb

```diff
@@ -149,68 +184,73 @@ def other_title(titles)
         # @return [String] the title value from combining the pieces of the structured_values by type and order
         #   with desired punctuation per specs
         #
+        # rubocop:disable Metrics/AbcSize
         # rubocop:disable Metrics/CyclomaticComplexity
         # rubocop:disable Metrics/MethodLength
         # rubocop:disable Metrics/PerceivedComplexity
-        def title_from_structured_values(title)
-          # parse out the parts
-          main_title = ''
-          subtitle = ''
-          non_sort_value = ''
+        def rebuild_structured_value(cocina_title)
+          result = ''
           part_name_number = ''
-          title.structuredValue.each do |structured_value|
-            # There can be a structuredValue inside a structuredValue.  For example,
+          cocina_title.structuredValue.each do |structured_value| # rubocop:disable Metrics/BlockLength
+            # There can be a structuredValue inside a structuredValue, for example,
             #   a uniform title where both the name and the title have internal StructuredValue
-            return title_from_structured_values(structured_value) if structured_value.structuredValue.present?
+            return rebuild_structured_value(structured_value) if structured_value.structuredValue.present?
```

Comment:
these changes look ugly, but they are not far off of the code I originally started with, before I mistakenly decided to reorder the pieces.